### PR TITLE
Decouple versionsController from Fedora 3

### LINKED ISSUE #2401: Refactor versions_controller to use dor-services-app metadata.version endpiont
TBD

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #2401

## How was this change tested?
Tested on QA.

### CODE REVIEW

File: app/controllers/versions_controller.rb [hunk 1]
```diff
@@ -10,15 +10,17 @@ def open_ui
...
 
   def close_ui
-    @description = @object.datastreams['versionMetadata'].current_description
-    @tag = @object.datastreams['versionMetadata'].current_tag
-
-    # do some stuff to figure out which part of the version number changed when opening
-    # the item for versioning, so that the form can pre-select the correct significance level
-    @changed_significance = which_significance_changed(get_current_version_tag(@object), get_prior_version_tag(@object))
+    versions = Dor::Services::Client.object(params[:item_id]).versions.inventory
+    previous_version, current_version = versions.sort(&:versionId).last(2)
```
Comment (mjgiarlo) [inline app/controllers/versions_controller.rb:14]: Are the versions not sorted by version ID already?

File: app/controllers/versions_controller.rb [hunk 2]
```diff
@@ -96,8 +77,8 @@ def which_significance_changed(cur_version_tag, prior_version_tag)
     nil
   end
 
-  # Filters
-  def create_obj
-    @object = Dor.find params[:item_id]
+  def load_and_authorize_resource
+    @cocina = Dor::Services::Client.object(params[:item_id]).find
```
Comment (mjgiarlo) [inline app/controllers/versions_controller.rb:81]: Is this always an item or can it vary? Either way, should we adopt our new naming scheme here and call this variable `@cocina_object` etc.?

File: spec/views/versions/_close_ui.html.erb_spec.rb [hunk 3]
```diff
@@ -3,11 +3,12 @@
 require 'rails_helper'
 
 RSpec.describe 'versions/_close_ui.html.erb' do
-  let(:object) { double('object', pid: 'druid:abc123') }
+  before do
+    @cocina = instance_double(Cocina::Models::DRO, externalIdentifier: 'druid:abc123')
```
Comment (mjgiarlo) [inline spec/views/versions/_close_ui.html.erb_spec.rb:7]: I'm curious why we need an ivar assignment here over, say, an `assign()` statement.

File: app/controllers/versions_controller.rb (hunk 1 above)
Comment (jcoyne) [reply app/controllers/versions_controller.rb:14]: This is what the code in dor-services is doing( https://github.com/sul-dlss/dor-services/blob/main/lib/dor/datastreams/version_metadata_ds.rb#L230) , so I just thought I ought to do it here, although they always do seem to be sorted.

File: spec/views/versions/_close_ui.html.erb_spec.rb (hunk 3 above)
Comment (jcoyne) [reply spec/views/versions/_close_ui.html.erb_spec.rb:7]: `assign()` is deprecated and using ivars is the preferred way.

File: app/controllers/versions_controller.rb (hunk 1 above)
Comment (mjgiarlo) [reply app/controllers/versions_controller.rb:14]: It's safe enough, nbd. No harm done in making the sort explicit on this side.

File: spec/views/versions/_close_ui.html.erb_spec.rb (hunk 3 above)
Comment (mjgiarlo) [reply spec/views/versions/_close_ui.html.erb_spec.rb:7]: Huh, OK! Good to know, thanks.

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**